ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/ethernet-interfaces.j2

```diff
@@ -165,21 +165,31 @@
 {%             endif %}
 {%         endfor %}
 {%     endif %}
-{# Transceiver Frequency #}
-{%     set custom_transceiver_frequency_interfaces = [] %}
+{# Transceiver Settings #}
+{%     set transceiver_settings = [] %}
 {%     for ethernet_interface in ethernet_interfaces | arista.avd.natural_sort('name') %}
-{%         if ethernet_interface.transceiver.frequency is arista.avd.defined %}
-{%             do custom_transceiver_frequency_interfaces.append(ethernet_interface) %}
+{%         if ethernet_interface.transceiver is arista.avd.defined %}
+{%             do transceiver_settings.append(ethernet_interface) %}
 {%         endif %}
 {%     endfor %}
-{%     if custom_transceiver_frequency_interfaces | length > 0 %}
-
-##### Transceiver Frequency
-
-| Interface | Transceiver Frequency |
-| --------- | --------------------- |
-{%         for custom_transceiver_frequency_interface in custom_transceiver_frequency_interfaces | arista.avd.natural_sort('name') %}
-| {{ custom_transceiver_frequency_interface.name }} | {{ custom_transceiver_frequency_interface.transceiver.frequency }} |
+{%     if transceiver_settings | length > 0 %}
+
+##### Transceiver Settings
+
+| Interface | Transceiver Frequency | Media Override |
+| --------- | --------------------- | -------------- |
+{%         for transceiver_setting in transceiver_settings | arista.avd.natural_sort('name') %}
+{%             set interface = transceiver_setting.name | arista.avd.default('-') %}
+{%             if transceiver_setting.transceiver.frequency is arista.avd.defined %}
+{%                 set frequency = "%.3f" | format(transceiver_setting.transceiver.frequency | float) %}
```

Comment:
I am no expert in format strings, but can we make it so it shows minimum 3 digits but more if they are not zeroes? As far as I can see on EOS that is how it behaves.

Reply:
let me check

Reply:
I don't think there is a convenient way of doing "3 or more" after the decimal point..

but telling from my test switch DCS-7280CR2A-30-F / 4.26.3M 

EOS will round the 4th decimal up or down and only display 3

```
(config-if-Et1/1)#transceiver frequency 190000.555555 ghz
(config-if-Et1/1)#sh run int et1/1
interface Ethernet1/1
   <snip>
   transceiver frequency 190000.556 ghz

(config-if-Et1/1)#transceiver frequency 190000.44444 ghz
(config-if-Et1/1)#sh run int et1/1
interface Ethernet1/1
   <snip>
   transceiver frequency 190000.444 ghz
```

did you test on a physical box or virtual?

Reply:
ok then keep the 3. I just found some internal docs that suggested smaller steps, but as long as we match the EOS behavior I am good :) Thanks.